freqtrade/exchange/__init__.py

```diff
@@ -48,390 +40,379 @@ def wrapper(*args, **kwargs):
-        _DRY_RUN_OPEN_ORDERS[order_id] = {
-            'pair': pair,
-            'price': rate,
-            'amount': amount,
-            'type': 'limit',
-            'side': 'sell',
-            'remaining': 0.0,
-            'datetime': arrow.utcnow().isoformat(),
-            'status': 'closed'
-        }
-        return {'id': order_id}
-
-    try:
-        return _API.create_limit_sell_order(pair, amount, rate)
-    except ccxt.InsufficientFunds as e:
-        raise DependencyException(
-            f'Insufficient funds to create limit sell order on market {pair}.'
-            f'Tried to sell amount {amount} at rate {rate} (total {rate*amount}).'
-            f'Message: {e}')
-    except ccxt.InvalidOrder as e:
-        raise DependencyException(
-            f'Could not create limit sell order on market {pair}.'
-            f'Tried to sell amount {amount} at rate {rate} (total {rate*amount}).'
-            f'Message: {e}')
-    except (ccxt.NetworkError, ccxt.ExchangeError) as e:
-        raise TemporaryError(
-            f'Could not place sell order due to {e.__class__.__name__}. Message: {e}')
-    except ccxt.BaseError as e:
-        raise OperationalException(e)
-
-
-@retrier
-def get_balance(currency: str) -> float:
-    if _CONF['dry_run']:
-        return 999.9
-
-    # ccxt exception is already handled by get_balances
-    balances = get_balances()
-    balance = balances.get(currency)
-    if balance is None:
-        raise TemporaryError(
-            f'Could not get {currency} balance due to malformed exchange response: {balances}')
-    return balance['free']
-
-
-@retrier
-def get_balances() -> dict:
-    if _CONF['dry_run']:
-        return {}
-
-    try:
-        balances = _API.fetch_balance()
-        # Remove additional info from ccxt results
-        balances.pop("info", None)
-        balances.pop("free", None)
-        balances.pop("total", None)
-        balances.pop("used", None)
-
-        return balances
-    except (ccxt.NetworkError, ccxt.ExchangeError) as e:
-        raise TemporaryError(
-            f'Could not get balance due to {e.__class__.__name__}. Message: {e}')
-    except ccxt.BaseError as e:
-        raise OperationalException(e)
-
-
-@retrier
-def get_tickers() -> Dict:
-    try:
-        return _API.fetch_tickers()
-    except ccxt.NotSupported as e:
-        raise OperationalException(
-            f'Exchange {_API.name} does not support fetching tickers in batch.'
-            f'Message: {e}')
-    except (ccxt.NetworkError, ccxt.ExchangeError) as e:
-        raise TemporaryError(
-            f'Could not load tickers due to {e.__class__.__name__}. Message: {e}')
-    except ccxt.BaseError as e:
-        raise OperationalException(e)
-
-
-@retrier
-def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
-    global _CACHED_TICKER
-    if refresh or pair not in _CACHED_TICKER.keys():
+class Exchange(object):
+
+    # Current selected exchange
+    _API: ccxt.Exchange = None
```

Comment:
I think those member variables should be lower case since they are no longer in global scope